### PR TITLE
Separate Public Router for Auth-Free Endpoints

### PR DESCRIPTION
**Pull request by kaxil**
Follow-up from https://github.com/apache/airflow/pull/43932
The PR splits the public_router by creating a common_router for routes that need 401/403 responses, separating them from routes without auth restrictions.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/__init__.py [hunk 1]
```diff
@@ -41,28 +41,34 @@
...
-public_router = AirflowRouter(
-    prefix="/public",
+public_router = AirflowRouter(prefix="/public")
+
+# Router with common attributes for all routes
+common_router = AirflowRouter(
     responses=create_openapi_http_exception_doc([status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN]),
 )
 
+common_router.include_router(assets_router)
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/__init__.py]: Should we have `authenticated_router` ? Maybe more explicit at first.

**Review by pierrejeambrun — APPROVED** [review]
Just one question, but already good to merge otherwise, thanks

File: airflow/api_fastapi/core_api/routes/public/__init__.py (hunk 1 above)
Comment (kaxil) [reply airflow/api_fastapi/core_api/routes/public/__init__.py]: Good idea

**Review by pierrejeambrun — APPROVED** [review]

**Comment by jscheffl** [comment]
Cool! Thanks!

**Pull request merged**